src/DefaultBuilder/src/WebApplicationBuilder.cs

```diff
@@ -63,21 +65,26 @@ internal WebApplicationBuilder(Assembly? callingAssembly, string[]? args = null,
             });
 
             // Apply the args to host configuration last since ConfigureWebHostDefaults overrides a host specific setting (the application name).
-            if (args is { Length: > 0 })
+
+            _bootstrapHostBuilder.ConfigureHostConfiguration(config =>
             {
-                _bootstrapHostBuilder.ConfigureHostConfiguration(config =>
+                if (args is { Length: > 0 })
                 {
                     config.AddCommandLine(args);
-                });
-            }
+                }
+
+                // Apply the options after the args
+                options.ApplyHostConfiguration(config);
+            });
+
 
             Configuration = new();
 
             // This is the application configuration
             var hostContext = _bootstrapHostBuilder.RunDefaultCallbacks(Configuration, _hostBuilder);
 
             // Grab the WebHostBuilderContext from the property bag to use in the ConfigureWebHostBuilder
-            var webHostContext = (WebHostBuilderContext)_hostBuilder.Properties[typeof(WebHostBuilderContext)];
+            var webHostContext = (WebHostBuilderContext)hostContext.Properties[typeof(WebHostBuilderContext)];
```

Comment:
I think `_hostBuilder.Properties` and `hostContext.Properties` have the same data at this point. Why use one over the other?

Reply:
There was no point to this change, I just like using hostContext here as a result of this call `_bootstrapHostBuilder.RunDefaultCallbacks` call.